twisted/web/test/test_http2.py

```diff
@@ -2079,6 +2079,64 @@ def validate(*args):
         return d.addCallback(validate)
 
 
+    def test_dataAndRstStream(self):
+        """
+        When a DATA frame is received at the same time as RST_STREAM,
+        Twisted does not send WINDOW_UPDATE frames for the stream.
+        """
+        frameFactory = FrameFactory()
+        transport = StringTransport()
+        a = H2Connection()
+        a.requestFactory = DummyHTTPHandler
+
+        # Send the request, but instead of the last frame send a RST_STREAM
+        # frame instead.
+        frames = buildRequestFrames(
+            headers=self.postRequestHeaders,
+            data=self.postRequestData,
+            frameFactory=frameFactory
+        )
+        del frames[-1]
+        frames.append(
+            frameFactory.buildRstStreamFrame(
+                streamID=1, errorCode=h2.errors.INTERNAL_ERROR
+            )
+        )
+
+        requestBytes = frameFactory.clientConnectionPreface()
+        requestBytes += b''.join(f.serialize() for f in frames)
```

Comment:
minor comment. I have seen these 2 lines repeated multiple times. maybe they can be refactored as

```
requestBytes = frameFactory.serializedClientFrames(frames)
```